ex3/k_namizaki/main3.py

```diff
@@ -0,0 +1,191 @@
+"""
+This code is to do regression analyzed using the least squares method.
+
+Calculate weights.
+Plot data.
+"""
+
+import argparse
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def parse_args():
+    """
+    Get Arguments.
+
+    Returns
+    -------
+    parser.parse_args() : 引数を返す
+    """
+    parser = argparse.ArgumentParser(description="最小二乗法を用いて回帰分析を行う。")
+    parser.add_argument(
+        "-file",
+        help="ファイルを入力",
+        default=r"C:\Users\kyskn\B4Lecture-2024\ex3\k_namizaki\data2.csv",
+        type=str,
+    )
+    parser.add_argument("-n", help="次数", default=1, type=int)
+    parser.add_argument("-normal", help="正則化係数", default=0, type=int)
+    return parser.parse_args()
+
+
+def plot2d(x, y, w):
+    """
+    Plot in 2 dimensions.
+
+    Parameters
+    ----------
+    x : xのデータ
+    y : yのデータ
+    w : 重み
```

Comment:
他の関数定義の箇所にも当てはまりますが，Docstringには型も書いておくと後々幸せになれるので推奨します．
いくつか主流のスタイルがありますが，GoogleスタイルかNumpyスタイルが使い勝手が良いと思います．[参考](https://qiita.com/flcn-x/items/393c6f1f1e1e5abec906)

Numpyスタイルで書くとこうなります．
データ型に，listやnumpy配列など，配列状のものを許容する場合はarray-likeと書くのが慣例だったと思います．
また，配列を引数に取る場合には，その形状を書いておくと良いです．2次元以上の配列を使う時，あとから混乱するのを防げます．

```suggestion
    Plot in 2 dimensions.

    Parameters
    ----------
    x (array-like): xのデータ. shape=(length)
    y (array-like): yのデータ. shape=(length)
    w (array-like): 重み. shape=(order)
```